examples/text/amazonreviewpolarity.py

```diff
@@ -37,25 +32,29 @@ def AmazonReviewPolarity(root, split):
     """Demonstrating caching, extraction and sanity check pipelines."""
 
     url_dp = IterableWrapper([URL])
-    # cache data on-disk with sanity check
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-
-    cache_dp = FileOpener(cache_dp, mode="b")
-
-    # stack TAR extractor on top of loader DP
-    extracted_files = cache_dp.read_from_tar()
-
-    # filter files as necessary
-    filter_extracted_files = extracted_files.filter(lambda x: split in x[0])
-
-    # stack sanity checker on top of extracted files
-    check_filter_extracted_files = filter_extracted_files.check_hash(
-        {os.path.normpath(os.path.join(root, _EXTRACTED_FILES[split])): _EXTRACTED_FILES_MD5[split]},
-        "md5",
-    )
-
-    # stack CSV reader and do some mapping
-    return check_filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), t[1]))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    def extracted_filepath_fn(x):
+        file_path = os.path.join(root, _EXTRACTED_FILES[split])
+        dir_path = os.path.dirname(file_path)
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        return file_path
+
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(
+        filepath_fn=extracted_filepath_fn)
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b").\
+                                            read_from_tar().\
+                                            filter(lambda x: _EXTRACTED_FILES[split] in x[0]).\
+                                            map(lambda x: (x[0].replace('_PATH' + '/', ''), x[1]))
```

Comment:
@ejguan   I think the path returned by read_from_tar is still `amazon_review_polarity_csv.tar.gz/amazon_review_polarity_csv/train.csv`. I have to remove `amazon_review_polarity_csv.tar.gz` to make it work properly.

Reply:
Ahh, actually ignore the part of "make it work properly". By using same_filepath_fn=True in end_caching, this part is irrelevant, thanks to @Nayef211 for catching that I am not using `_PATH` properly. That said, the extracted filename still contains `amazon_review_polarity_csv.tar.gz`, this is not really a concern though :)

Reply:
I think we can get rid of the whole map operation here.